src/components/TokenInfo/index.tsx

```diff
@@ -0,0 +1,113 @@
+  const useUSDCPrice = useUSDCPriceHook[chainId];
+  const useTotalSupply = useTotalSupplyHook[chainId];
+
+  const balance = useTokenBalance(account ?? undefined, token);
+
+  const tokenPrice = useUSDCPrice(token);
+
+  const chainTotalSuppply = useTotalSupply(token);
+
+  const _totalSupply = totalSupply ?? chainTotalSuppply;
+
+  const totalAccountBalance = balance && unclaimedAmount ? balance?.add(unclaimedAmount) : new TokenAmount(token, '0');
+
+  return (
+    <Wrapper>
+      <Text color="text1" lineHeight="20px" fontWeight={500}>
+        {t('header.pngBreakDown', { symbol: token?.symbol })}
+      </Text>
+      <Frame>
+        {logo ? (
+          <StyledLogo
+            size="48px"
+            srcs={[logo]}
+            alt={`${token.symbol} Logo`}
+            className={animatedLogo ? 'Animated' : undefined}
+          />
+        ) : (
+          <CurrencyLogo currency={token} size={48} className={animatedLogo ? 'Animated' : undefined} />
+        )}
+        {account && (
+          <Text color="text1" fontSize={32} fontWeight={600} data-tip data-for="totalAccountBalance-tip">
+            {`${numeral(totalAccountBalance?.toFixed(2) ?? '0').format('0.00a')}`}
+          </Text>
+        )}
+        <Tooltip id="totalAccountBalance-tip" effect="solid" backgroundColor={theme.primary}>
+          <Text color="eerieBlack" fontSize="12px" fontWeight={500} textAlign="center">
+            {totalAccountBalance.toFixed(token.decimals)} {token?.symbol}
+          </Text>
+        </Tooltip>
+      </Frame>
+      <Frame>
+        {account && (
+          <TextInfo
+            text={t('header.balance')}
+            value={`${numeral(balance?.toFixed(2) ?? '0').format('0.00a')}`}
+            tooltipText={balance?.toFixed(token.decimals) ?? '0.00'}
+          />
+        )}
+        {unclaimedAmount && <TextInfo text={t('header.unclaimed')} value={unclaimedAmount.toFixed(2)} />}
+      </Frame>
+      <Frame>
+        <TextInfo
+          text={t('header.tokenPrice', { symbol: token?.symbol })}
+          value={`$ ${numeral(tokenPrice?.toFixed(2) ?? '0').format('0.00a')}`}
+        />
+        {circulationSupply && (
+          <TextInfo
+            text={t('header.pngCirculation', { symbol: token?.symbol })}
+            value={`${numeral(circulationSupply.toFixed(2) ?? '0').format('0.00a')}`}
+          />
+        )}
+        <TextInfo
+          text={t('header.totalSupply')}
+          value={`${numeral(_totalSupply?.toFixed(2) ?? '0').format('0.00a')}`}
+        />
+      </Frame>
+    </Wrapper>
+  );
+}
+
+export function TokenInfoModal({ open, closeModal, ...rest }: TokenInfoModalProps) {
+  return (
+    <Modal isOpen={open} onDismiss={closeModal}>
+      <Box width="420px" position="relative">
+        <TokenInfo {...rest} />
+        <Box position="absolute" right="20px" top="15px">
+          <CloseButton onClick={closeModal} size={16} />
+        </Box>
+      </Box>
+    </Modal>
+  );
+}
```

Comment:
lets move this into separate file TokenInfoModal.tsx as we try to not keep multiple components in same file.